pkg/provider/kubernetes/crd/kubernetes.go

```diff
@@ -509,6 +510,49 @@ func buildTLSOptions(ctx context.Context, client Client) map[string]tls.Options
 	return tlsOptions
 }
 
+func buildTLSStores(ctx context.Context, client Client) map[string]tls.Store {
+	tlsStoreCRD := client.GetTLSStores()
+	var tlsStores map[string]tls.Store
+
+	if len(tlsStoreCRD) == 0 {
+		return tlsStores
+	}
+	tlsStores = make(map[string]tls.Store)
+
+	for _, tlsStore := range tlsStoreCRD {
+		namespace := tlsStore.Namespace
+		logger := log.FromContext(log.With(ctx, log.Str("tlsStore", tlsStore.Name), log.Str("namespace", namespace)))
+		secretName := tlsStore.Spec.DefaultCertificate.SecretName
+
+		secret, exists, err := client.GetSecret(namespace, secretName)
+		if err != nil {
+			logger.Errorf("Failed to fetch secret %s/%s: %v", namespace, secretName, err)
+			return tlsStores
+		}
+		if !exists {
+			logger.Errorf("Secret %s/%s does not exist", namespace, secretName)
+			return tlsStores
+		}
+
+		cert, key, err := getCertificateBlocks(secret, namespace, secretName)
+		if err != nil {
+			logger.Errorf("Could not get certificate blocks: %v", err)
+			return tlsStores
+		}
+
+		if _, ok := tlsStores["default"]; !ok {
+			// Since only the default TLS Store is used, we just need to set it if it doesn't exist.
+			tlsStores["default"] = tls.Store{
+				DefaultCertificate: &tls.Certificate{
+					CertFile: tls.FileOrContent(cert),
+					KeyFile:  tls.FileOrContent(key),
+				},
+			}
+		}
```

Comment:
So, just to be clear:
we allow for the CRDs to contain several TLS stores, but we only take into account the first valid one we find while iterating through them?
If yes, why do we go through the trouble of iterating through all of them then?
Why do not simply change line 543 to:

    if _, ok := tlsStores["default"]; ok {
        break
    }

?

Reply:
Why not?

Because I'm not as awesome as you are.

<3